compiler/src/dmd/location.d

```diff
@@ -38,19 +33,19 @@ debug info etc.
 */
 struct Loc
 {
-    private uint _linnum;
-    private uint _charnum;
-    private uint fileIndex; // index into filenames[], starting from 1 (0 means no filename)
-    version (LocOffset)
-        uint fileOffset; /// utf8 code unit index relative to start of file, starting from 0
+    private uint index = 0; // offset into lineTable[]
+
+    // FIXME: This arbitrary size increase is needed to prevent segfault in
+    // runnable/test42.d on Ubuntu x86 when DMD was built with DMD 2.105 .. 2.110
+    // https://github.com/dlang/dmd/pull/20777#issuecomment-2614128849
+    version (linux) version (DigitalMars) static if (size_t.sizeof == 4)
```

Comment:
```suggestion
    version (DigitalMars) version (linux) version (X86)
```
Firstly because version conditions are cheaper than static if, secondly because `size_t == 4` includes X32, or any other 32bit pointer abi.